libvast/src/detail/mmapbuf.cpp

```diff
@@ -45,29 +45,42 @@ mmapbuf::mmapbuf(size_t size)
   setp(map_, map_ + size_);
 }
 
+mmapbuf::mmapbuf(const int fd, const size_t size, const size_t offset)
+  : is_file_owner_{true},
```

Comment:
Why does the overload that passes a file descriptor imply ownership? Shouldn't it be the opposite? In this version, you're calling `std::remove` in the destructor, which doesn't make much sense to me.

Reply:
This is a mistake, thanks for pointing it out.